website/newsletter.py

```diff
@@ -0,0 +1,149 @@
+import os
+import json
+import hashlib
+import requests
+from config import config
+from functools import wraps
+from hedy_content import COUNTRIES
+from website.auth import send_email
+
+
+MAILCHIMP_API_URL = None
+MAILCHIMP_API_HEADERS = {}
+if os.getenv("MAILCHIMP_API_KEY") and os.getenv("MAILCHIMP_AUDIENCE_ID"):
+    # The domain in the path is the server name, which is contained in the Mailchimp API key
+    MAILCHIMP_API_URL = (
+        "https://"
+        + os.getenv("MAILCHIMP_API_KEY").split("-")[1]
+        + ".api.mailchimp.com/3.0/lists/"
+        + os.getenv("MAILCHIMP_AUDIENCE_ID")
+    )
+    MAILCHIMP_API_HEADERS = {
+        "Content-Type": "application/json",
+        "Authorization": "apikey " + os.getenv("MAILCHIMP_API_KEY"),
+    }
+
+
+def run_if_mailchimp_config_present(f):
+    """Similar to 'requires_login', but also tests that the user is an admin."""
+    @wraps(f)
+    def inner(*args, **kws):
+        if MAILCHIMP_API_URL:
+            return f(*args, **kws)
```

Comment:
Super nitpick: probaby we shouldn't be doing `return f`: `f` can only ever return `None` because it may be that we're not calling it at all, in which case we will *definitely* return `None` so you can't rely on a return value anyway?